react/src/components/ServiceLauncherPageContent.tsx

```diff
@@ -128,12 +134,15 @@
 
   const webuiNavigate = useWebUINavigate();
   const baiClient = useSuspendedBackendaiClient();
+  const baiRequestWithPromise = useBaiSignedRequestWithPromise();
   const currentDomain = useCurrentDomainValue();
 
   const [isOpenServiceValidationModal, setIsOpenServiceValidationModal] =
     useState(false);
 
-  const formRef = useRef<FormInstance<ServiceLauncherFormValue>>(null);
+  const [form] = Form.useForm<ServiceLauncherFormValue>();
+  const runtimeVariant = Form.useWatch('runtimeVariant', form);
+  const vFolderName = Form.useWatch('vFolderName', form);
```

Comment:
`useWatch` should be the last option to handle form data changes because it triggers a re-render of the current component every time it changes. It's an opposite way to handle form data in an uncontrolled manner.

I pushed [this code](https://github.com/lablup/backend.ai-webui/pull/2472/commits/ac4c8aaed07df7f03c9d5a8b654530bee2451380#diff-14075520518ef75d88c4d5df326da07601b2eb8721afadb0d0c9629306b63284R793) in my previous commit to handle form value dependency without using `useWatch`.